src/azul/service/avro_pfb.py

```diff
@@ -112,13 +112,11 @@ def add_doc(self, doc: JSON):
         assert contents['files'] is not doc['contents']['files']
         file_relations = set()
         for entity_type, entities in contents.items():
-            # FIXME: Protocol entities lack document ID so we skip for now
-            #        https://github.com/DataBiosphere/azul/issues/3084
-            entities = (e for e in entities if 'document_id' in e)
             # Sorting entities is required for deterministic output since
             # the order of the inner entities in an aggregate document is
             # tied to the order with which contributions are returned by ES
             # during aggregation, which happens to be non-deterministic.
+            entities = (e for e in entities if 'document_id' in e)
```

Comment:
The comment above is about sorting which is done in the line below. So I don't think this line should be moved.